clang/include/clang/Interpreter/Interpreter.h

```diff
@@ -137,9 +136,10 @@ class Interpreter {
 
   Expr *SynthesizeExpr(Expr *E);
 
-private:
   size_t getEffectivePTUSize() const;
+  void finalizeInitPTUStack();
```

Comment:
We could also consider renaming this to `getNumUndoSteps()` and `setUndoBarrier()` respectively (or some thing like that).

Reply:
Maybe something like `markRestorePoint` or similar?

Reply:
Hm, why restore point? Do you mean sth like `markMaxRestorePoint` as in "the last point we can revert to"? And would you keep `getEffectivePTUSize` for the getter then? It seemed reasonable to me to make a relation between the two. I still think restoration is a misleading term, because we can only recover the compiler state precisely. Execution state is not restorable in the same way.

What else does it affect apart from Undo? If it's only Undo, then I'd be inclined to have it in the name. If we explicitly don't want a relation between then two, it could also just be `finalize`.

Reply:
> Do you mean sth like markMaxRestorePoint as in "the last point we can revert to"? 

Yes, this is the point the user code starts and there is no point to undo more because we will start reverting builtins. Maybe we can include user code in the name... Like `markUserCodeStart` or something.